src/DssSpell.t.sol

```diff
@@ -667,69 +738,30 @@ contract DssSpellTest is DssSpellTestBase {
-    }
-
-    function testRWA009DocChange() public {
-        checkRWADocUpdate("RWA009-A", OLD_RWA009_DOC, NEW_RWA009_DOC);
-    }
-
-    function testRWA007OraclePriceBump() public {
-        (, address pip, , ) = liquidationOracle.ilks("RWA007-A");
-        (,,uint256 spot,,) = vat.ilks("RWA007-A");
-
-        assertEq(uint256(DSValueAbstract(pip).read()), 250 * MILLION * WAD, "RWA007: Bad initial PIP value");
-        assertEq(spot, 250 * MILLION * RAY, "RWA007: Bad initial spot value");
-
-        vote(address(spell));
-        scheduleWaitAndCast(address(spell));
-        assertTrue(spell.done());
-
-        (uint256 Art, uint256 rate, uint256 spotAfter, uint256 line,) = vat.ilks("RWA007-A");
-
-        assertEq(uint256(DSValueAbstract(pip).read()), 500 * MILLION * WAD, "RWA007: Bad PIP value after bump()");
-        assertEq(spotAfter, 500 * MILLION * RAY, "RWA007: Bad spot value after bump()");
-
-        // Test that a draw can be performed.
-        address urn = addr.addr("RWA007_A_URN");
-        giveAuth(urn, address(this));
-        RwaUrnLike(urn).hope(address(this));  // become operator
-        uint256 room = sub(line, mul(Art, rate));
-        uint256 drawAmt = room / RAY;
-        if (mul(divup(mul(drawAmt, RAY), rate), rate) > room) {
-            drawAmt = sub(room, rate) / RAY;
-        }
-        RwaUrnLike(urn).draw(drawAmt);
-        (Art,,,,) = vat.ilks("RWA007-A");
-        assertTrue(sub(line, mul(Art, rate)) < mul(2, rate));  // got very close to line
-    }
-
-    // CHANGELOG Houskeeping
-    function testChangelogHousekeeping() public {
-        address rwa007inUrn = chainLog.getAddress("RWA007_A_INPUT_CONDUIT_URN");
-        address rwa007inJar = chainLog.getAddress("RWA007_A_INPUT_CONDUIT_JAR");
-
+    function testDirectCompV2Integration() public {
         vote(address(spell));
         scheduleWaitAndCast(address(spell));
         assertTrue(spell.done());
 
-        assertEq(chainLog.getAddress("RWA007_A_INPUT_CONDUIT"), rwa007inUrn);
-        assertEq(chainLog.getAddress("RWA007_A_JAR_INPUT_CONDUIT"), rwa007inJar);
+        bytes32 ilk = "DIRECT-COMPV2-DAI";
+        D3MHubLike hub = D3MHubLike(addr.addr("DIRECT_HUB"));
+        address pool = addr.addr("DIRECT_COMPV2_DAI_POOL");
+        address plan = addr.addr("DIRECT_COMPV2_DAI_PLAN");
+        D3MMomLike mom = D3MMomLike(addr.addr("DIRECT_MOM"));
+        
+        // Current market conditions should max out the D3M @ 5m DAI
+        hub.exec(ilk);
+        (uint256 ink, uint256 art) = vat.urns(ilk, pool);
+        assertEq(ink, 5 * MILLION * WAD);
+        assertEq(art, 5 * MILLION * WAD);
+
+        // De-activate the D3M via mom
+        hevm.prank(pauseProxy);
+        mom.disable(plan);
+        assertEq(D3MCompoundPlanLike(plan).barb(), 0);
+        hub.exec(ilk);
+        (ink, art) = vat.urns(ilk, pool);
+        assertLt(ink, WAD);     // Less than some dust amount is fine (1 DAI)
+        assertLt(art, WAD);
```

Comment:
We probably want to explicitly check all the values related to the d3m that we file in the spell (hub[ilk].pool, hub[ilk].plan, hub[ilk].tau, hub.vow, hub.end, pool.king, plan.barb, oracle.hub, spot.pip). Might have missed something so please double check.

Also relying of the hub on the vat.

spot.mat is covered  in config.sol already.

Reply:
Ya good point. Added a bunch of checks.